src/util/path.ts

```diff
@@ -2,26 +2,133 @@ import * as vscode from 'vscode';
 import * as path from 'path';
 import untildify = require('untildify');
 
-/**
- * Given relative path, calculate absolute path.
- */
-export function GetAbsolutePath(partialPath: string): string {
-  const editorFilePath = vscode.window.activeTextEditor!.document.uri.fsPath;
-  let basePath: string;
-
-  if (partialPath.startsWith('/')) {
-    basePath = '/';
-  } else if (partialPath.startsWith('~/')) {
-    basePath = <string>untildify(partialPath);
-    partialPath = '';
-  } else if (partialPath.startsWith('./')) {
-    basePath = path.dirname(editorFilePath);
-    partialPath = partialPath.replace('./', '');
-  } else if (partialPath.startsWith('../')) {
-    basePath = path.dirname(editorFilePath) + '/';
+export function separatePath(searchPath: string, separator: string) {
```

Comment:
can you add a function header/comment explaining this function? especially as you are returning multiple things, we should document what's being returned and the order.

Reply:
Added